### DISCUSSION TITLE
Hook into legacy xamarin forms application

### DISCUSSION THREAD

**Discussion by Moret84** [shinyorg/shiny#1383] [Q&A]
Hi, thanks for your work.

I am attempting to hook shiny inside a legacy Xamarin forms application but I am confused about which section of the documentation I should use: https://shinylib.net/client/hosting/native/ or https://shinylib.net/client/hosting/manual/ 

I have tried both but in the former case, I can't call my LoadApplication(new App()) from a ShinyAndroidActivity since it does not inherit from FormsAppCompatActivity and in the latter case I have kept my MainActivity as is, adding only an Application but it my app hangs forever and it seems OnCreate is never called at all

The application has a pile of Init inside its MainActivity and has also a SplashActivity that seems to run MainActivity. It is really a mess. Have I any chance to success hooking shiny in that ?

Many thanks for your help

**Comment by aritchie** [comment]
IF you have an existing application on XF with a lot of existing calls (Inits, etc) - you want the manual hosting option.  It just requires you manually calling the lifecycle stuff of Shiny (more inits).
